### ISSUE TITLE
Applying selection to opacity no longer respect default opacity

### ISSUE THREAD

**Issue by kanitw** [vega/vega-lite#4302]
[Example](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEAkDODGALApgWwIaQFxUQFzwAdYsB6UgN2QHN0A6agSz0QFcAjOxge1IRQyUa6ALQAbZskoBmOgCtY3AHaQANOCgATZAgBOjQnh4qckACK701MN1Z4w6MLuTw86JdVZj0usO12ssIhgeNxgiIzUiBJR9oTcjEp4sHRqGpCa6G7YYMCQrLpiORlZ6KTwPikKypAAvuoQkLDIYi5GNTigEI3+gYg5Xd2NeACehMjFiXjIuhToRRoQtRr16Ri6ANbF8VNpjchK8NyaidQDi1AAHgOQAGaMLZrFABLcus3xAO4zalCj48UAI6sdxGNxGKh1BrdSAjG73R7FACyjFasAA+uNdOiAOLzMQ1VR-MYTUzA0HMLKMSGrIaQI4E3Twh5iJ6mADCIwkSm071+kH+pKgbxOSnmUIukG4hHQ8GYcM6F0aRx5zGM5yGdOmsDwNyU3F1nSaLTa6tMvSCdXqYCVMLmYlYQoADHQAEy25ZDT1gZbLIA)

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v3.json",
  "description": "Drag out a rectangular brush to highlight points.",
  "data": {"url": "data/cars.json"},
  "selection": {
    "brush": {
      "type": "interval"
    }
  },
  "mark": "point",
  "encoding": {
    "x": {"field": "Horsepower", "type": "quantitative"},
    "y": {"field": "Miles_per_Gallon", "type": "quantitative"},
    "color": {"field": "Cylinders", "type": "ordinal"},
    "opacity": {
      "condition": {
        "test": {"not": {"selection": "brush"}}, 
        "value": 0.2
      }
    }
  }
}
```

The opacity is no longer `0.7` by default.

**Comment by kanitw** [comment]
The bug is in `mixins.ts`

```js
export function baseEncodeEntry(model: UnitModel, ignore: Ignore) {
  const {fill, stroke} = color(model);
  return {
    ...markDefProperties(model.markDef, ignore),
    ...wrapInvalid(model, 'fill', fill),
    ...wrapInvalid(model, 'stroke', stroke),
    ...nonPosition('opacity', model),
    ...nonPosition('fillOpacity', model),
    ...nonPosition('strokeOpacity', model),
    ...nonPosition('strokeWidth', model),
    ...tooltip(model),
    ...text(model, 'href')
  };
}
```

Basically, the default opacity logic is in `markDefProperties`, but it will be overridden by `nonPosition('opacity', model)`.

**Comment by kanitw** [comment]
Fixed in #4303
